### PR TITLE
Support an NVHPC config

### PR DESCRIPTION
**Pull request by wmaxey**
There appears to be a deadlock in semaphore when compiling with  NVC++. It's tricky and happens occasionally.

In order to not completely kill testing, I've cobbled together the timeout stuff into our config.

### CODE REVIEW

**Review by miscco — APPROVED** [review]
Please double check, there were some incoming `_LIBCUDACXX_COMPILER_PGI` recently

**Comment by wmaxey** [comment]
> Please double check, there were some incoming `_LIBCUDACXX_COMPILER_PGI` recently

Do you mean there are more? I replaced everything I found.

**Comment by miscco** [comment]
> > Please double check, there were some incoming `_LIBCUDACXX_COMPILER_PGI` recently
> 
> Do you mean there are more? I replaced everything I found.

I merged main and replaced the newly added occurences

File: .upstream-tests/test/support/test_macros.h [hunk 1]
```diff
@@ -62,7 +62,7 @@
 #define TEST_HAS_BUILTIN_IDENTIFIER(X) 0
 #endif
 
-#if defined(__PGIC__)
+#if defined(__NVHPC__)
 # define TEST_COMPILER_PGI
```
Comment (miscco) [inline .upstream-tests/test/support/test_macros.h:66]: I think you need to replace that one too ^^

File: include/cuda/std/detail/libcxx/include/cmath [hunk 2]
```diff
@@ -308,6 +308,10 @@ long double    truncl(long double x);
 #include <math.h>
 #endif
 
+#if defined(__cuda_std__) && defined(_LIBCUDACXX_COMPILER_NVHPC)
```
Comment (wmaxey) [inline include/cuda/std/detail/libcxx/include/cmath:311]: https://clang.godbolt.org/z/hMWf98ae1

There is some weirdness depending on the toolchain NVC++ uses. Not sure how we need to configure the NVHPC image to get it to emit some of these functions.

Comment (miscco) [reply include/cuda/std/detail/libcxx/include/cmath:311]: I am strongly against us having to work around missing functionality. We should rather ask where those can be found.

Comment (wmaxey) [reply include/cuda/std/detail/libcxx/include/cmath:311]: Fixed after consultation with David.

**Pull request merged**